### ISSUE TITLE
Development tokens not working after 4.23.0

### ISSUE THREAD

**Issue by vankien96** [GetStream/stream-chat-swift#2382]
## What did you do?
Just update from 4.22.0 to 4.23.0

## What did you expect to happen?
Not crash

## What happened instead?
It crashed when I call connect function
<img width="553" alt="image" src="https://user-images.githubusercontent.com/26648523/203017734-3631422e-9242-4869-ab7c-75d30c9ce4eb.png">

<img width="1391" alt="image" src="https://user-images.githubusercontent.com/26648523/203017345-4f087bf1-e187-490d-9a5d-6c7e7cdb5cb4.png">

## GetStream Environment
**GetStream Chat version:** 4.23.0
**GetStream Chat frameworks:** StreamChat, StreamChatUI
**iOS version:** Simulator 16.0
**Swift version:**
**Xcode version:**14.0
**Device:** Simulator

**Comment by nuno-vieira** [comment]
Hi @vankien96!

Thank you for the report! We will tackle this one as soon as possible.

Btw, what is your use case for using`ChatConnectionController` directly?

Best,
Nuno

**Comment by vankien96** [comment]
Hi @nuno-vieira , 
This is what I call when initializing StreamChat. I wanted to know when StreamChat finishes init, after that I will update user information.
<img width="846" alt="image" src="https://user-images.githubusercontent.com/26648523/203074795-e1ad36ac-e56d-491f-b0ba-e952a5cb21ea.png">

**Comment by nuno-vieira** [comment]
Hi @vankien96 

There is no need for that. You can update the user info directly in the connectUser function. Whatever you set in the UserInfo data, it will update the user. There's no need to observe the connection status. 

Let us know if that unblocked you for the meantime.

Best,
Nuno

**Comment by vankien96** [comment]
Thanks @nuno-vieira, I removed `chatClient?.connectionController().connect` and it doesn't crash anymore.
btw, It works well before updating to 4.23, so I think it will be good for your side to know what has changed and make it crash.

**Comment by vankien96** [comment]
Hi @nuno-vieira , Could you help me on this?
After updating to 4.24 and removing `chatClient?.connectionController().connect`
I got these error:
<img width="592" alt="image" src="https://user-images.githubusercontent.com/26648523/203221742-2458494d-9876-4f1b-8405-26c3b7a58dbb.png">
<img width="595" alt="image" src="https://user-images.githubusercontent.com/26648523/203221768-7838dd7b-cf7d-4334-b681-20f846e73b79.png">

Here is my setup:
<img width="840" alt="image" src="https://user-images.githubusercontent.com/26648523/203221829-ea6baf91-4e09-46a5-98d7-6d2de8e2f9de.png">
<img width="745" alt="image" src="https://user-images.githubusercontent.com/26648523/203221894-c2d51886-4f9a-4817-a6ee-f2d7e29b428f.png">

**Comment by nuno-vieira** [comment]
Hi @vankien96, what is your `updateUser()` function doing? As I said before, you should put user data in the `connect(userInfo:)` property.  It will update the user data while connecting to it.

**Comment by vankien96** [comment]
Hi @nuno-vieira ,
It just update name and imageURL
<img width="662" alt="image" src="https://user-images.githubusercontent.com/26648523/203313045-3753b836-63df-4c5c-ae4c-28448bdb9dc8.png">

**Comment by nuno-vieira** [comment]
Hi @vankien96,

Why don't you do this instead?

```swift
connectUser(
         userInfo: .init(
            id: streamUserId,
            name: fullName,
            imageURL: URL(string: currentUser?.photoPath ?? ""),
            extraData: [ChatUser.auth0Id: .string(auth0Id)]
        )
```
With the code above, you connect and update the user at the same time, with just one call.

Best,
Nuno

**Comment by vankien96** [comment]
Hi @nuno-vieira , 
Ah yes, but i think it's not something will cause these error above. 
![image](https://user-images.githubusercontent.com/26648523/203368153-33f6f433-a730-4483-886b-d59d2bb0fff1.png)

**Comment by vankien96** [comment]
Here you go @nuno-vieira 

```
curl 'https://chat.stream-io-api.com/channels?api_key=jtkdhxxwcq73&payload=%7B%22member_limit%22:30,%22message_limit%22:25,%22filter_conditions%22:%7B%22$and%22:%5B%7B%22members%22:%7B%22$in%22:%5B%22auth06332b9d89aad072b4c9717f3%22%5D%7D%7D,%7B%22last_message_at%22:%7B%22$gt%22:%221970-01-01T00:00:00Z%22%7D%7D%5D%7D,%22limit%22:20,%22watch%22:true%7D' \
-H 'Host: chat.stream-io-api.com' \
-H 'Accept: */*' \
-H 'User-Agent: Dev-Supermomos/1 CFNetwork/1390 Darwin/22.1.0' \
-H 'Accept-Language: vi-VN,vi;q=0.9' \
-H 'X-Stream-Client: stream-chat-uikit-client-v4.24.0|app=dev.com.supermomos|app_version=1.54|os=iOS 16.0|device_model=arm64|device_screen_ratio=3.00' \
-H 'Connection: keep-alive' \
--proxy http://localhost:9090
```

**Comment by nuno-vieira** [comment]
Hi @vankien96!

Can you share your current setup now with the latest changes?

Thanks!

**Comment by nuno-vieira** [comment]
@vankien96 Was this issue happening as well before moving the UserInfo details to the connectUser call?

**Comment by vankien96** [comment]
Hi @nuno-vieira,
This issue only happens from 4.23. I'm still using 4.22 and it works well without any change in connectUser.

**Comment by nuno-vieira** [comment]
@vankien96, when using the development token, did you disable the auth checks? As described here: https://getstream.io/chat/docs/sdk/ios/guides/working-with-user/#development-tokens:~:text=You%20can%20only,for%20production%20applications.

**Comment by vankien96** [comment]
Hi @nuno-vieira , Yes we have already disabled it.

**Comment by nuno-vieira** [comment]
Hi @vankien96,

After analysing in-depth on your issue, now I fully understand what is happening. Your code snippet is actually a misuse of the SDK, and the crash was intended. You were basically connecting the user 2x. Since you called `connectUser() `and then `connect()` with the `ConnectionController`. What you actually want is the snippet I gave to you, where you update the user in the connectUser() call, and at most, you could also do this:
```swift
connectUser {
   updateUser()
}
```

But now the issue you are having is actually that we broke the development tokens in 4.24.0. This is now being fixed here: https://github.com/GetStream/stream-chat-swift/pull/2393

Sorry for the inconvenience! We will let you know once this is released.

Best,
Nuno

**Comment by vankien96** [comment]
Wow, nice. Thanks @nuno-vieira

**Comment by polqf** [comment]
Hi! This has been fixed in our latest release [4.25.0](https://github.com/GetStream/stream-chat-swift/releases/tag/4.25.0)
Closing this issue for now, feel free to reopen it if you encounter any issues.
